config-provisioning/src/main/java/com/yahoo/config/provision/ZoneId.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.config.provision;
+
+import java.util.Objects;
+
+/**
+ * Unique identifier for a Zone; use when referencing them.
+ *
+ * Serialised form is 'environment.region'.
+ *
+ * @author jvenstad
+ */
+public class ZoneId {
+
+    protected final Environment environment;
```

Comment:
No big deal since they are final, but why not make these private?

Reply:
Both this, and the `@Override`, was because I used IntelliJ to do most of the code refactoring. I'll check again what other things I may have forgotten to clean up. 
(I first extracted ZoneId as a superclass, and changed all usages in controller code to the superclass, and then removed the subclassing again. Some things were left dangling.)